src/data/padics/padic_norm.lean

```diff
@@ -298,6 +316,225 @@ theorem min_le_padic_val_rat_add {q r : ℚ}
 
 end padic_val_rat
 
+section padic_val_nat
+
+/--
+Dividing out by a prime factor reduces the padic_val_nat by 1.
+-/
+lemma padic_val_nat_div {p : ℕ} [p_prime : fact p.prime] {b : ℕ} (dvd : p ∣ b) :
+  (padic_val_nat p (b / p)) = (padic_val_nat p b) - 1 :=
+begin
+  by_cases b_split : (b = 0),
+  { simp [b_split], },
+  { have split_frac : padic_val_rat p (b / p) = padic_val_rat p b - padic_val_rat p p :=
+      padic_val_rat.div p (nat.cast_ne_zero.mpr b_split) (nat.cast_ne_zero.mpr (nat.prime.ne_zero p_prime)),
+    rw padic_val_rat.padic_val_rat_self (nat.prime.one_lt p_prime) at split_frac,
+    have r : 1 ≤ padic_val_nat p b := one_le_padic_val_nat_of_dvd b_split dvd,
+    exact_mod_cast split_frac, }
+end
+
+/--
+If a prime doesn't appear in `n`, `padic_val_nat p n` is `0`.
+-/
+lemma padic_val_nat_of_not_dvd {p : ℕ} [fact p.prime] {n : ℕ} (not_dvd : ¬(p ∣ n)) :
+  padic_val_nat p n = 0 :=
+begin
+  by_cases hn : n = 0,
+  { subst hn, simp at not_dvd, trivial, },
+  { rw padic_val_nat_def hn,
+    exact (@multiplicity.unique' _ _ _ p n 0 (by simp) (by simpa using not_dvd)).symm,
+    assumption, },
+end
+
+lemma padic_val_nat_primes {p q : ℕ} [p_prime : fact p.prime] [q_prime : fact q.prime] (neq : p ≠ q) :
+  padic_val_nat p q = 0 :=
+@padic_val_nat_of_not_dvd p p_prime q $ (not_congr (iff.symm (prime_dvd_prime_iff_eq p_prime q_prime))).mp neq
+
+/--
+Dividing out by a prime which is not a factor doesn't change the `padic_val_nat`.
+-/
+-- The strange naming order here, with the two-tick version appearing before the one-tick version,
+-- is because this is a special case of the one-tick version which is nevertheless of interest in
+-- its own right.
+lemma padic_val_nat_div'' {p : ℕ} [p_prime : fact p.prime] {q : ℕ} [q_prime : fact q.prime] (neq : p ≠ q)
+  {b : ℕ} (dvd : q ∣ b) : padic_val_nat p (b / q) = padic_val_nat p b :=
+begin
+  by_cases b_split : (b = 0),
+  { subst b,
+    simp [padic_val_nat], },
+  { have q_nonzero : q ≠ 0,
+    { intros q_zero,
+      simpa [q_zero, zero_dvd_iff] using dvd, },
+    have e : padic_val_rat p (b / q) = padic_val_rat p b - padic_val_rat p q :=
+      padic_val_rat.div p (nat.cast_ne_zero.mpr b_split) (nat.cast_ne_zero.mpr q_nonzero),
+    rw [← padic_val_rat_of_nat p q, padic_val_nat_primes neq] at e,
+    simp only [int.coe_nat_zero, sub_zero] at e,
+    have cz : char_zero ℚ := linear_ordered_semiring.to_char_zero,
+    rw [←padic_val_rat_of_nat p b, ←@cast_dvd_char_zero _ _ cz _ _ dvd, ←padic_val_rat_of_nat p (b / q)] at e,
+    exact_mod_cast int.coe_nat_inj e, },
+end
+
+@[simp]
+lemma div_div_div {a b c : ℕ} (dvd : c ∣ a) (dvd2 : a ∣ b) : (b / (a / c)) / c = b / a :=
```

Comment:
I guess you're already aware of this, but this should of course move to some `nat.basic` file or such

Reply:
I couldn't really think of a name for it. Do you have any suggestions?

Reply:
Not really, maybe ask on Zulip.